internal/app/connectconformance/results.go

```diff
@@ -159,6 +159,9 @@ func (r *testResults) failed(testCase string, err *conformancev1.ClientErrorResu
 func (r *testResults) assert(testCase string, expected, actual *conformancev1.ClientResponseResult) {
 	var errs multiErrors
 
+	errs = append(errs, checkError(expected.Error, actual.Error)...)
+	errs = append(errs, checkPayloads(expected.Payloads, actual.Payloads)...)
+
```

Comment:
This was a slight UX improvement -- often, the reason a test case fails is because an unexpected error occurs. Previously when that happened, the test case would complain about response headers and trailers before getting to the "meat" of the problem: the error. So this reorders the checks so it is more likely the most relevant errors are reported first.